v4/examples/langchain_langsmith.mdx

```diff
@@ -0,0 +1,428 @@
+            "end": "2024-04-15T19:45:00.000Z"
+        },
+	    ....
+        {
+            "start": "2024-04-19T18:45:00.000Z",
+            "end": "2024-04-19T21:00:00.000Z"
+        }
+    ],
+    "oooExcludedDateRanges": [
+
+    ],
+    "workingHours": [
+        {
+            "days": [
+                1,
+                2,
+                3,
+                4,
+                5
+            ],
+            "startTime": 780,
+            "endTime": 1260,
+            "userId": xxxx
+        }
+    ],
+    "dateOverrides": [],
+    "currentSeats": null,
+    "datesOutOfOffice": {}
+}
+```
+
+Great! Now we know that our API key is working and pulling information from our calendar. The API calls we will be using later in this tutorial are:
+
+- **/availability**: Get your availability
+- **/bookings**: Book a slot
+
+### Cerebrium setup
+
+If you don’t have a Cerebrium account, you can create one by signing up [here](https://dashboard.cerebrium.ai/register) and following the documentation [here](https://docs.cerebrium.ai/cerebrium/getting-started/installation) to get setup
+
+In your IDE, run the following command to create our Cerebrium starter project: cerebrium init agent-tool-calling. This creates two files:
+
+- Main.py - Our entrypoint file where our code lives
+- cerebrium.toml - A configuration file that contains all our build and environment settings
+  ‍
+
+Add the following pip packages near the bottom of your cerebrium.toml. This will be used in creating our deployment environment.
+
+```
+[cerebrium.dependencies.pip]
+pydantic = "latest"
+langchain = "latest"
+pytz = "latest" ##this is used for timezones
+openai = "latest"
+langchain_openai = "latest"
+```
+
+We will be using OpenAI GPT3.5 for our use cases and so we need an API key from them. If you don’t have an account, you can sign up [here](https://openai.com/). You can then create an API key [here](https://platform.openai.com/settings/profile?tab=api-keys). The API key should be in the format: “sk_xxxxx”.
+
+In your Cerebrium dashboard you can then add your Cal.com and OpenAI API keys as secrets by navigating to “Secrets” in the sidebar. For the sake of this tutorial I called mine “CAL_API_KEY” and “OPENAI_API_KEY”. We can now access these values in our code at runtime without exposing them in our code.
+
+![Cerebrium Secrets Dashboard](/images/examples/langchain_langsmith/cerebrium_secrets.png)
+
+### Agent Setup
+
+To start we need to write two tool functions that the agent will use to check availability on our calendar as well as book a slot.
+
+- Get availability tool
+- You would have seen from the test API request we did above to Cal.com that the API returns your availability in the following way:
+- The time slots that you are already busy
+- Your working hours on each day
+  ‍
+
+Below is the code to achieve this
+
+```python
+
+from langchain_core.tools import tool
+from cerebrium import get_secret
+import requests
+from cal import find_available_slots
```

Comment:
You don't actually show the user how to do this in the example (They have to navigate to the repo to find out). This is disconcerting as you only figure this out when you actually try to execute the main python file locally much lower down in the example

Reply:
I explain it just below that